go/vt/sqlparser/sql.y

```diff
@@ -2178,6 +2180,10 @@ char_type:
   {
     $$ = &ColumnType{Type: string($1), EnumValues: $3, Charset: $5}
   }
+| VECTOR length_opt
+  {
+    $$ = &ColumnType{Type: string($1), Length: $2}
+  }
```

Comment:
This is the new column type definition that is possible with `VECTOR`. We can add it safely here to the parser, since even if we can parse it here, the underlying MySQL will then error with a syntax error. That is the same behavior as today regardless, since if we have non-properly parsed DDL, we already pass it down as a best effort to MySQL anyway. 

So a syntax error today that the user sees already comes from MySQL and that doesn't change. In general also, our Vitess parser is more lax than MySQL itself. It's reverse engineered from MySQL and in case of doubt we always opt for accepting a bit more than MySQL would if we can't easily seem to match its behavior. That way we don't block valid queries and invalid queries are tried but then throw a syntax error regardless from MySQL itself.

Reply:
This could be updated later, once the capabilities code knows about vector?